### DISCUSSION TITLE
Create a new file based on the current file's title

### DISCUSSION THREAD

**Discussion by byerun** [SilentVoid13/Templater#1000] [Help] [answered]
I would like to be able to have Templater create a new based on the title of the current file. For example, the current file might be named "Can't log into computer XYZ" and I want to then have Templater create a file named "Can't log into computer XYZ - address" with content

```
problem:: [[Can't log into computer XYZ]]

# Solution
```

Is this possible using Templater?

**Comment by welpdx** [comment, marked as answer]
woohooo discussion #1000! 

Try this script:
```js
<%* 
//get title and appends text to the end
const nameVar = tp.file.title + " - " + "address"
//searches for the [[template]] template to add template content
const templateVar = tp.file.find_tfile("template")
//create new file with filename and content from [[template]]
await tp.file.create_new(templateVar, nameVar)
%>
```

In [[template]]:
```md

%% 
tp.file.title gets the title. 
split() gets the first half before the hyphen. 
trim() removes whitespaces. 
tR is the output, thats why we use \<\%* 
%%
problem: [[<%* tR += tp.file.title.split("-")[0].trim()%>]]

# Solution
```

<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/TFQ4Pcd.gif" alt="demo" />
</details>


~welp

**Reply by byerun** [reply]
Thank you @welpdx! The missing piece for me was usage of the [tR](https://silentvoid13.github.io/Templater/commands/execution-command.html?#how-to-output-a-value-from-a-javascript-execution-command-) string.

**Reply by welpdx** [reply]
@byerun awesome to hear that this was helpful!
If I have answered your original question, please mark this post as answered! 
Thank you!

**Reply by byerun** [reply]
Done!

**Reply by welpdx** [reply]
Thanks!
